File: Travel-hub/Travelhub.swift/Models/Friend.swift
//
//  Friend.swift
//  Travelhub
//
//  Created by Jonas Groll on 11.02.26.
//

import Foundation
import SwiftData

@Model
final class Friend {
    var id: String = UUID().uuidString
    var friendName: String
    var friendEmail: String
    var status: String = "accepted" // "pending", "accepted", "blocked"
    var addedDate: Date = Date()
    var profileImage: String = "person.circle.fill"
    
    init(friendName: String, friendEmail: String, status: String = "pending") {
        self.friendName = friendName
        self.friendEmail = friendEmail
        self.status = status
    }
}

File: travelhub_flutter/lib/models/friend.dart
import 'package:uuid/uuid.dart';

const _uuid = Uuid();

class Friend {
  final String id;
  String friendName;
  String friendEmail;
  String status; // "pending", "accepted", "blocked"
  DateTime addedDate;
  String profileImage;

  Friend({
    String? id,
    required this.friendName,
    required this.friendEmail,
    this.status = 'pending',
    DateTime? addedDate,
    this.profileImage = 'person.circle.fill',
  })  : id = id ?? _uuid.v4(),
        addedDate = addedDate ?? DateTime.now();

  Map<String, dynamic> toMap() => {
        'id': id,
        'friendName': friendName,
        'friendEmail': friendEmail,
        'status': status,
        'addedDate': addedDate.toIso8601String(),
        'profileImage': profileImage,
      };

  factory Friend.fromMap(Map<String, dynamic> m) => Friend(
        id: m['id'] as String,
        friendName: m['friendName'] as String,
        friendEmail: m['friendEmail'] as String,
        status: (m['status'] ?? 'pending') as String,
        addedDate: DateTime.parse(m['addedDate'] as String),
        profileImage: (m['profileImage'] ?? '') as String,
      );
}
